zaptest/observer/observer.go

```diff
@@ -116,7 +123,9 @@ func (o *ObservedLogs) FilterFieldKey(key string) *ObservedLogs {
 	})
 }
 
-func (o *ObservedLogs) filter(match func(LoggedEntry) bool) *ObservedLogs {
+// Filter filters entries to those for which the provided function returns
+// true.
+func (o *ObservedLogs) Filter(match func(LoggedEntry) bool) *ObservedLogs {
```

Comment:
minor: consider renaming `match` to `keep` to make it clear that entries that this function returns true for will be kept in the returned ObservedLogs.